packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -305,12 +317,21 @@ class CustomPaint extends SingleChildRenderObjectWidget {
   /// instead.
   final Size size;
 
+  /// Whether the painting is complex enough to benefit from caching.
```

Comment:
this would be a great place to copy the informative paragraph you wrote for RenderCustomPaint (this is more likely to be seen than the other). Duplicating the docs is fine.